operations/host_provision.go

```diff
@@ -87,6 +113,29 @@ func hostProvision() cli.Command {
 	}
 }
 
+func postHostIsUp(ctx context.Context, comm client.Communicator, hostID, hostSecret, cloudProvider string) (*restmodel.APIHost, error) {
+	var ec2InstanceID string
+	if cloud.IsEC2InstanceID(hostID) {
```

Comment:
We can't because it's the app server that doesn't know the host exists, and it's easier if we know all hosts ping the app server to indicate the host is up rather than have to conditionally do it (the host lifecycle is full of conditionals, so having fewer asterisks attached to each step simplifies things). Anyways, saving API calls is a negligible concern since making one network request on its own is cheap - it's the logic that the app server does (if any) that's expensive.